### PR TITLE
change check on start timestamp

### PR DESCRIPTION
**Pull request by dwisth**
@mcamurri 

This changes the check on the timestamps in the undistorter.

This fixes an edge case where we try to undistort the point cloud to a time between the point cloud header time, and the time of the first point in the cloud (see diagram below).

image

### CODE REVIEW

File: lidar_undistortion/include/lidar_undistortion/lidar_undistorter.hpp [hunk 1]
```diff
@@ -131,10 +131,13 @@ inline bool LidarUndistorter<PointT>::processCloud(const typename PointCloud::Pt
                                  << beams_ << " time look up table size is "
                                  << times_lut_.size());
   DEBUG_PRINTLN("minmax times : " << (*minmax_time.first ) << " " << (*minmax_time.second));
-  uint64_t t_start = start_timestamp + (*minmax_time.first );
-  uint64_t t_end =   start_timestamp + (*minmax_time.second);
+  uint64_t t_start = start_timestamp + static_cast<uint64_t>(*minmax_time.first );
+  uint64_t t_end =   start_timestamp + static_cast<uint64_t>(*minmax_time.second);
```
Comment (wxmerkt) [inline lidar_undistortion/include/lidar_undistortion/lidar_undistorter.hpp]: This appears to be a fix for a bug that could be caught when turning on all warnings more aggressively by default and fixing the arising warnings: `-Wall -Wpedantic -Wextra`

Comment (dwisth) [reply lidar_undistortion/include/lidar_undistortion/lidar_undistorter.hpp]: True - this should be picked up by those warnings. The types are `uint64_t` and `int32_t` and the first variable `uint64_t` can be a very large number (the ros timestamp in nanoseconds).

This wasn't the main bug that I was trying to fix but should be resolved.

**Comment by mcamurri** [comment]
In no case `start_timestamp` and `t_start` should be different. If this happens, it means the point cloud was badly timestamp in the first place, so the problem should be fixed at the acquisition of the point cloud and not here. I agree there should be here an extra check that `start_timestamp` and `t_start` should be identical and it should fire an error if this is not a case.

Regarding the static casting, it should be something like this:
```cpp
uint64_t t_start = static_cast<uint64_t>(static_cast<int64_t>(start_timestamp) + static_cast<int64_t>(*minmax_time.first))
```

**Review by mcamurri — CHANGES_REQUESTED** [review]
I would add the static casting as shown in my previous comment

**Pull request merged**